app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee.erb

```diff
@@ -505,9 +505,14 @@ ShortFormApplicationService = (
   Service.signInSubmitApplication = (opts = {}) ->
     # check if this user has already applied to this listing
     Service.getMyAccountApplication().success((data) ->
-      if !_.isEmpty(data.application) && Service._previousIsSubmittedOrBothDrafts(data.application)
+      previousApplication = data.application
+      if !_.isEmpty(previousApplication)
         # if user already had an application for this listing
-        return Service._signInAndSkipSubmit(data.application)
+        if opts.type == 'review-sign-in' && previousApplication.status.match(/draft/i)
+          # override draft ID and proceed, so that we eventually call submitApplication...
+          Service.overridePreviousDraftId()
```

Comment:
@chrisdolendo so yeah the same `overridePreviousDraftId` is used for choosing a draft, however in this case they have finished/reviewed their latest draft and are ready to submit, so it is sort of counterproductive to say "hey you had an old draft, do you want to use it?" since that one is probably less finished than this one, and they'd have to start back over. So in this case we basically silently overwrite their old draft (without asking them) so that they still only have _one_ application but it is replaced with this newly finished one. (we checked with Ashley that this is how we should proceed)

Will add a better clarifying comment in the code than the current one.